### PR TITLE
mesa: 22.1.7 -> 22.2.1

### LINKED ISSUE #193655: Mesa 22.2.0
###### Checklist
https://docs.mesa3d.org/relnotes/22.2.0.html

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: legacyPackages.x86_64-linux.mesa

current version: 22.1.7
desired version: 22.2.0

###### Notify maintainers

maintainers: @primeos @vcunat 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by PedroHLC**
###### Description of changes

- Bumped version;
- Rebased patches;
- Remove upstreamed patches;
- Includes https://github.com/NixOS/nixpkgs/pull/192243#issuecomment-1261242034
- Fixes https://github.com/NixOS/nixpkgs/issues/193655

###### Things done

Built `mesa`, `pkgsMusl.mesa`, `pkgsi686Linux.mesa`.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -5,9 +5,10 @@
 , llvmPackages, libffi, libomxil-bellagio, libva-minimal
 , libelf, libvdpau
 , libglvnd, libunwind
-, vulkan-loader
+, vulkan-loader, glslang
 , galliumDrivers ? ["auto"]
 , vulkanDrivers ? ["auto"]
+, videoCodecs ? [ "h264dec" "h264enc" "h265dec" "h265enc" "vc1dec" ]
```
Comment (PedroHLC) [inline pkgs/development/libraries/mesa/default.nix:11]: @jtojnar what do you think, list or `withPatentedCodecs ? true`?

File: pkgs/development/libraries/mesa/musl-stacksize.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
+From: https://github.com/void-linux/void-packages/blob/b9f58f303ae23754c95d5d1fe87a98b5a2d8f271/srcpkgs/mesa/patches/musl-stacksize.patch
```
[the remark this answers is not in the document]
Comment (PedroHLC) [reply pkgs/development/libraries/mesa/musl-stacksize.patch:1]: Good catch.

**Comment by PedroHLC** [comment]
Thanks for the base changing @vcunat, I totally forgot it this time.

**Comment by vcunat** [comment]
It didn't work out, as right now it contains (many) commits from `staging..master`.   It would probably suffice to wait for 4 hours due to auto-merges happening.

**Pull request closed without merging**